profiling/src/profile/internal/sample.rs

```diff
@@ -1,16 +1,17 @@
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
 
-use super::{Label, StackTraceId};
+use super::{LabelId, LabelSetId, StackTraceId};
 use std::hash::Hash;
 #[derive(Eq, PartialEq, Hash)]
 pub struct Sample {
     /// label includes additional context for this sample. It can include
     /// things like a thread id, allocation size, etc
-    pub labels: Vec<Label>,
+    pub labels: LabelSetId,
```

Comment:
Minor: Should this perhaps be renamed `label_set`?

Reply:
Personally, I find this name cleaner.